File: remohouse-core/src/main/java/org/circuitdoctor/core/service/ActionLogGSMService.java
package org.circuitdoctor.core.service;

import org.circuitdoctor.core.model.ActionLogGSM;
import org.circuitdoctor.core.model.GSMController;
import org.circuitdoctor.core.model.User;

import java.util.List;
import java.util.Set;

public interface ActionLogGSMService {
    ActionLogGSM addActionLogGSM(ActionLogGSM actionLogGSM);
    Set<ActionLogGSM> findAllActions(Long userID);
    void deleteActionsWithUser(User user);
    void deleteActionsWithGSMController(GSMController gsmController);
}

File: front-end/src/app/service/ActionLogGSMService.ts
import {Injectable} from "@angular/core";
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Observable} from "rxjs";
import {ActionLogGSM} from "../model/ActionLogGSM";

@Injectable({providedIn: 'root'})
export class ActionLogGSMService {

  private url = 'http://localhost:8080/api/actionLogGSM';

  constructor(private http: HttpClient) {
  }
  private httpSecurity = {
    headers: new HttpHeaders({
      Authorization : `Basic ${window.btoa('user:password')}`
    })
  };

  getActions(userID : number) : Observable<Array<ActionLogGSM>>{
    return this.http.get<Array<ActionLogGSM>>(this.url+"/getActions/"+userID, this.httpSecurity);
  }

}
